test/gleam/regex_test.gleam

```diff
@@ -42,6 +42,22 @@ pub fn check_test() {
 
   regex.check(re, "boo")
   |> should.be_false
+
+  // On target JavaScript state is carried over for global regexp:
```

Comment:
Not happy with the wording, but without an explanation why the seeminly stupid tests below are there I'd not know why they are there and just delete them eventually?